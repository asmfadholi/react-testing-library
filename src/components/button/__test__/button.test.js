 import React from 'react';
 import ReactDom from 'react-dom'; 
 import Button from '../button.js';

 import { render, cleanup, fireEvent } from '@testing-library/react';
//  import '@testing-library/jest-dom'; 
 import '@testing-library/jest-dom/extend-expect'
 import renderer from 'react-test-renderer'

 afterEach(cleanup)

 it('Renders without crashing', () => {
     const div = document.createElement("div")
     ReactDom.render(<Button></Button>, div)
     ReactDom.unmountComponentAtNode(div)
 })

 it('Renders Button correctly', () => {
    const { getByTestId } = render(<Button label="Click me"></Button>)
    expect(getByTestId('button')).toHaveTextContent("Click me");
 })

 it('Click button correctly', () => {
    const { getByTestId } = render(<Button label="Click me"></Button>)
    const countTextNode = getByTestId("count");
    fireEvent.click(countTextNode)

    expect(countTextNode).toHaveTextContent('Click to increase: 1');
    fireEvent.click(countTextNode)
    expect(countTextNode).toHaveTextContent('Click to increase: 2');
 })

 it('Rerenders Button correctly', () => {
    const { getByTestId, rerender } = render(<Button label="Click me"></Button>)
    rerender(<Button label="Push"></Button>)
    expect(getByTestId('button')).toHaveTextContent("Push");
 })

 it('Matched snapshot 1', () => {
    const tree = renderer.create(<Button label="save"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
 })

 it('Matched snapshot 2', () => {
    const tree = renderer.create(<Button label="Click me"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
 })