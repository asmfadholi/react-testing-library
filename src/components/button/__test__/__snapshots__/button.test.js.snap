// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matched snapshot 1 1`] = `
Array [
  <div
    className="button-style"
    data-testid="button"
  >
    save
  </div>,
  <button
    data-testid="count"
    onClick={[Function]}
  >
    Click to increase: 
    0
  </button>,
]
`;

exports[`Matched snapshot 2 1`] = `
Array [
  <div
    className="button-style"
    data-testid="button"
  >
    Click me
  </div>,
  <button
    data-testid="count"
    onClick={[Function]}
  >
    Click to increase: 
    0
  </button>,
]
`;
